### PR TITLE
Update rival and fav opponent

### LINKED ISSUE #85: Slow Run Time in favorite_opponent and rival
The favorite_opponent and rival methods within team_statistics.rb run slower than desired. 

Please look over and find out where we get more efficiency.

### PR DESCRIPTION
**Pull request by ryandbarnett**
I believe that I reduced the number of loops required but it is still a large method.

It passes both the harness and our tests.

### CODE REVIEW

File: lib/team_statistics.rb [hunk 1]
```diff
@@ -118,11 +118,65 @@ def favorite_opponent(team_id)
...
+  #       win_percent
+  #   end
+  #   find_team(lowest_win_percentage[0].to_s).teamname
+  # end
+
   def rival(team_id)
-    lowest_win_percentage = win_percent_per_team(team_id).max_by do |t_id, win_percent|
-        win_percent
+    # sort the hash to find rival
+    rival_id = opponent_win_count(team_id).sort_by {|k, v| v}.last[0]
```
Comment (dhalverson) [inline lib/team_statistics.rb]: I like this method, I'm curious though at the time difference between the original and the updated one?

**Review by dhalverson — COMMENTED** [review]
Do you recall the time difference in yours vs the original?

**Comment by ryandbarnett** [comment]
The time difference is 7-8 secs vs 48 seconds

**Review by hashmaster3k — APPROVED** [review]
Great refactor!

**Comment by ryandbarnett** [comment]
Turns out I needed to use the percentage just like the instructions said! The main changes I made to get that to work are....

- In the sort_by in rival and fav opponent methods the value's (which is a Hash now not an Interger) average
- In the opponent win count helper method the initial value is an object with the keys wins, total, and average
- In change opponent count helper the wins, total, and average is changed instead of count

File: lib/team_statistics.rb [hunk 2]
```diff
@@ -76,53 +77,59 @@ def fewest_goals_scored(team_id)
...
+  def favorite_opponent(team_id)
+    rival_id = opponent_win_count(team_id).sort_by { |k, v| v[:average] }.first[0]
+    rival_team = find_team(rival_id.to_s)
+    rival_team.teamname
   end
 
-  def get_games_by_id(team_id)
-    get_all_team_games(team_id).map do |game|
-      game.game_id
+  def rival(team_id)
```
Comment (dhalverson) [inline lib/team_statistics.rb:86]: This looks good, we reviewed on a call in person, and this def runs much faster than the previous way.

**Review by dhalverson — APPROVED** [review]
This is huge improvement from the previous method, approved

**Pull request merged**